src/main/java/org/wso2/integration/connector/core/AbstractConnector.java

```diff
@@ -28,106 +30,131 @@
+
+    public boolean mediate(MessageContext messageContext) {
+
+        try {
+            connect(messageContext);
+        } catch (ConnectException e) {
+            throw new SynapseException("Error occurred when connecting conenctor. Details :", e);
+        }
+        return true;
+    }
+
+    protected Object getParameter(MessageContext messageContext, String paramName) {
+
+        return ConnectorUtils.lookupTemplateParamater(messageContext, paramName);
+    }
+
+    protected <T> T getMediatorParameter(
+            MessageContext messageContext, String parameterName, Class<T> type, boolean isOptional) {
+
+        Object parameter = getParameter(messageContext, parameterName);
+        if (!isOptional && (parameter == null || parameter.toString().isEmpty())) {
+            handleException(String.format("Parameter %s is not provided", parameterName), messageContext);
+        } else if (parameter == null || parameter.toString().isEmpty()) {
+            return null;
+        }
+
+        try {
+            return parse(Objects.requireNonNull(parameter).toString(), type);
+        } catch (IllegalArgumentException e) {
+            handleException(String.format(
+                    "Parameter %s is not of type %s", parameterName, type.getName()
+                                         ), messageContext);
+        }
+
+        return null;
+    }
+
+    protected <T> T getProperty(
+            MessageContext messageContext, String propertyName, Class<T> type, boolean isOptional) {
+
+        Object property = messageContext.getProperty(propertyName);
+        if (!isOptional && (property == null || property.toString().isEmpty())) {
+            handleException(String.format("Property %s is not set", propertyName), messageContext);
+        } else if (property == null || property.toString().isEmpty()) {
+            return null;
+        }
+
+        try {
+            return parse(Objects.requireNonNull(property).toString(), type);
+        } catch (IllegalArgumentException e) {
+            handleException(String.format(
+                    "Property %s is not of type %s", propertyName, type.getName()
+                                         ), messageContext);
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T parse(String value, Class<T> type) throws IllegalArgumentException {
+
+        if (type == Integer.class) {
+            return (T) Integer.valueOf(value);
+        } else if (type == Double.class) {
+            return (T) Double.valueOf(value);
+        } else if (type == Boolean.class) {
+            return (T) Boolean.valueOf(value);
+        } else if (type == String.class) {
+            return (T) PayloadUtils.removeQuotesIfExist(value);
+        } else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
+        }
+    }
+
+    protected void handleConnectorResponse(MessageContext messageContext, String responseVariable,
```

Comment:
Shall we merge these two methods into one? They seem identical except for the type of payload. It seems like having only the second implementation is enough if you convert the String payload into a JsonElement payload from where the method is being invoked.

Reply:
We are having two methods here because from the connector implementation we can either have a Json object or a String value. As per the current runtime implementation, we have to pass a string if we are writing to the synapse message body. If we have to store it in the variable, we have to store it as a json object.